### PR TITLE
Added connection_init message

### PR DESCRIPTION
**Pull request by Amzd**
#32

### CODE REVIEW

File: Sources/SwiftGraphQL/HTTP.swift [hunk 1]
```diff
@@ -247,7 +252,17 @@ private func listen<Type, TypeLock>(
...
     request.httpBody = nil
     let socket: URLSessionWebSocketTask = URLSession.shared.webSocketTask(with: request)
-
+    
+    // GQL_CONNECTION_INIT
+    // Client sends this message after plain websocket connection to start the communication with the server
+    let connectionInit: Data = try! JSONSerialization.data(withJSONObject: ["type": "connection_init"])
+    socket.send(.data(connectionInit)) { error in
+        if let error = error {
+            print("GQL_CONNECTION_INIT error:", error)
```
Comment (maticzav) [inline Sources/SwiftGraphQL/HTTP.swift:261]: is there a way to get rid of the print statement?

File: Sources/SwiftGraphQL/HTTP.swift [hunk 2]
```diff
@@ -232,11 +232,16 @@ private func listen<Type, TypeLock>(
...
             case let .success(message):
                 // Try to serialize the response.
-                if let data = message.data, let result = try? GraphQLResult(webSocketResponse: data, with: selection) {
-                    eventHandler(.success(result))
-                } else {
-                    eventHandler(.failure(.badpayload))
-                }
+                if let data = message.data {
+                    let jsonObject = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+                    if jsonObject?["type"] as? String == "data" {
```
Comment (maticzav) [inline Sources/SwiftGraphQL/HTTP.swift:237]: Could you paste an example return object that the socket sends? It seems like we could create a `GraphQLSubscriptionResponse` structure that contains the `type` property and decode it there.

Does that make sense to you?

**Review by maticzav — CHANGES_REQUESTED** [review]
Hey!

Thanks! You are great. This library wouldn't be half as great without you.

File: Sources/SwiftGraphQL/HTTP.swift (hunk 2 above)
Comment (Amzd) [reply Sources/SwiftGraphQL/HTTP.swift:237]: Yep we need to change this anyway because we'll start using the id field there too when we fix #33

**Comment by Amzd** [comment]
This doesn't have connectionParams which are needed for auth with apollo so I'm closing this for now. Will be fixed in new PR.

**Pull request closed without merging**